src/TestFramework/Microsoft.Rest.ClientRuntime.Azure.TestFramework/TestEnvironment.cs

```diff
@@ -322,29 +348,59 @@ private void InteractiveLogin(string tenant, ActiveDirectoryServiceSettings aadS
         /// </summary>
         private void VerifySubscription()
         {
-#region
-            if (!(string.IsNullOrEmpty(this.SubscriptionId)) || (this.SubscriptionId.Equals("None", StringComparison.OrdinalIgnoreCase)))
+            #region
+            string matchedSubscriptionId = string.Empty;
+            StringBuilder sb = new StringBuilder();
+            string callerId = string.Empty;
+            string subs = string.Empty;
+
+            if (this.TokenInfo[TokenAudience.Management] != null)
             {
-                //TokenCredentials cred = this.TokenInfo[TokenAudience.Management];
+                try { callerId = this.TokenInfo[TokenAudience.Management].CallerId; } catch { }
```

Comment:
especially during interactive login, we actually set the connection string userId field and then show up in an exception message that this particular user that was provided interactively does not have associated subscription that was provided in connection string.
This helps with more information for the user to act upon if for some reason an incorrect subscription was provided in connection string.